core/src/main/scala/org/apache/spark/deploy/DependencyUtils.scala

```diff
@@ -123,6 +123,11 @@ private[deploy] object DependencyUtils {
 
     uri.getScheme match {
       case "file" | "local" => path
+      case "http" | "https" | "ftp" if Utils.isTesting =>
```

Comment:
@vanzin , it is a little difficult to mock the download behavior, so here I check if "spark.testing" is configured, return a dummy local path if it is configured. What do you think about this approach?

Reply:
Sounds good.